libs/data-access/api/src/lib/trpc/routers/learning-diary.router.ts

```diff
@@ -171,32 +173,89 @@ export const learningDiaryPageRouter = t.router({
 			);
 		}
 
+		// get location ID (available to everyone(?)) TEMPORAL FIX
+		let locationID;
+		if (input.learningLocation) {
+			// Try to find personalized case
+			const existingLocation = await database.learningLocation.findUnique({
+				where: {
+					unique_name_creator: {
+						name: input.learningLocation.name,
+						creatorName: ctx.user.name
+					}
+				},
+				select: {
+					id: true
+				}
+			});
+			if (existingLocation) {
+				locationID = existingLocation.id;
+			} else {
+				// Try to find generalized case
+				const existingLocation = await database.learningLocation.findFirst({
+					where: {
+						name: input.learningLocation.name,
+						creatorName: null
+					},
+					select: {
+						id: true
+					}
+				});
+				if (existingLocation) {
+					locationID = existingLocation.id;
+				}
+			}
+		}
```

Comment:
prisma lets you combine the database query into one using "OR". 

```suggestion
let locationID: string | undefined;

if (input.learningLocation) {
	const existingLocation = await database.learningLocation.findFirst({
		where: {
			OR: [
				{
					unique_name_creator: {
						name: input.learningLocation.name,
						creatorName: ctx.user.name
					}
				},
				{
					name: input.learningLocation.name,
					creatorName: null
				}
			]
		},
		select: {
			id: true
		},
		orderBy: {
			creatorName: 'desc' // Prefer personalized entry over general one
		}
	});

	if (existingLocation) {
		locationID = existingLocation.id;
	}
}

```

Reply:
findFirst can not work with unique_name_creator and findUnique does not accept OR. There is another way, but it does not rely on indexed fields:
...
where: {
    name: input.learningLocation.name,
    OR: [
      { creatorName: null },
      { creatorName: ctx.user.name }
    ]
  },
 ...
 
 Availability of two locations with the same name (and different order) can lead to annoying bug where user selects general location, but it selects personalized. This issue must be solved elsewhere

Reply:
![image](https://github.com/user-attachments/assets/ddf4901d-7a59-4ec7-bd64-4899d843b038)